src/sage/matroids/chow_ring.py

```diff
@@ -466,3 +514,37 @@ def homogeneous_degree(self):
             if not f.is_homogeneous():
                 raise ValueError("element is not homogeneous")
             return f.degree()
+
+        def _acted_upon_(self, scalar, self_on_left=True):
+            r"""
+            Return the action of ``scalar`` on ``self``.
+
+            EXAMPLES::
+
+                sage: SGA = SymmetricGroupAlgebra(GF(3), 4)
+                sage: GP22 = SGA.garsia_procesi_module([2, 2])
+                sage: x = SGA.an_element(); x
+                [1, 2, 3, 4] + 2*[1, 2, 4, 3] + [4, 1, 2, 3]
+                sage: v = GP22.an_element(); v
+                -gp1 - gp2 - gp3
+                sage: g = SGA.group().an_element(); g
+                [4, 1, 2, 3]
+                sage: g * v  # indirect doctest
+                gp3
+                sage: x * v  # indirect doctest
+                gp3
+                sage: 2 * v  # indirect doctest
+                gp1 + gp2 + gp3
+            """
+            P = self.parent()
+            if scalar in P.base_ring():
+                return super()._acted_upon_(scalar, self_on_left)
+            if scalar in P._matroid.automorphism_group():
+                gens = P.ambient().gens()
+                return P.retract(self.lift().subs({g: gens[scalar(i+1)-1] for i, g in enumerate(gens)}))
+            if not self_on_left and scalar in P._matroid.automorphism_group():
+                scalar = P._semigroup_algebra(scalar)
```

Comment:
```suggestion
            if scalar in P._matroid.automorphism_group():
                gens = P.ambient().gens()
                return P.retract(self.lift().subs({g: gens[scalar(i+1)-1] for i, g in enumerate(gens)}))
            if not self_on_left and scalar in P._semigroup_algebra:
                scalar = P._semigroup_algebra(scalar)
```
Although I don't think `_semigroup_algebra` has been defined yet. You also don't have a test for something in the (semi)group algebra acting.